src/compiler/commandLineParser.ts

```diff
@@ -405,13 +405,106 @@ namespace ts {
       */
     export function parseConfigFileTextToJson(fileName: string, jsonText: string): { config?: any; error?: Diagnostic } {
         try {
-            return { config: /\S/.test(jsonText) ? JSON.parse(jsonText) : {} };
+            let jsonTextWithoutComments = removeComments(jsonText);
+            return { config: /\S/.test(jsonTextWithoutComments) ? JSON.parse(jsonTextWithoutComments) : {} };
         }
         catch (e) {
             return { error: createCompilerDiagnostic(Diagnostics.Failed_to_parse_file_0_Colon_1, fileName, e.message) };
         }
     }
 
+
+    /**
+     * Remove the comments from a json like text.
+     * Comments can be single line comments (starting with # or //) or multiline comments using / * * /
+     *
+     * This method replace comment content by whitespace rather than completely remove them to keep positions in json parsing error reporting accurate.
+     */
+    function removeComments(jsonText: string): string {
+        let result = "";
+        let processingString = false;
+        let processingSingleLineComment = false;
+        let processingMultiLineComment = false;
+        for (let i = 0; i < jsonText.length; i++) {
+            let currentChar = jsonText.charAt(i);
+            let nextChar = (i + 1 < jsonText.length) ? jsonText.charAt(i + 1) : undefined;
+            if (processingString) {
+                if (currentChar === "\\"

```

Comment:
i would rather you use charCodes and use ts.CharacterCodes for all these comparisons. this way things are more consistent.